tests/test_warpedvrt.py

```diff
@@ -399,3 +399,14 @@ def test_warpedvrt_float32_preserve(data):
     with rasterio.open("tests/data/float32.tif") as src:
         with WarpedVRT(src, src_crs="EPSG:4326") as vrt:
             assert src.dtypes == vrt.dtypes == ("float32",)
+
+
+def test_warpedvrt_float32_override(data):
+    """Override GDAL defaults for working data type"""
+    float32file = str(data.join("float32.tif"))
+    with rasterio.open(float32file, "r+") as dst:
+        dst.nodata = -3.4028230607370965e+38
+
+    with rasterio.open(float32file) as src:
+        with WarpedVRT(src, src_crs="EPSG:4326", dtype="float32") as vrt:
+            assert src.dtypes == vrt.dtypes == ("float32",)
```

Comment:
And here's the confirmation of the fix.

Reply:
@sgillies I didn't check the file but to make sure it actually works can you add `nodata=0.0001` in the parameter as shown in https://github.com/mapbox/rasterio/issues/1744#issuecomment-528510998

Reply:
Done.